File: FinanceApp.Web/src/models/IncomeTransactionDtos/GetIncomeTransactionDto.ts
import { GetIncomeTransactionGroupDto } from "./GetIncomeTransactionGroupDto";
import { Money } from "../Money/Money";

export interface GetIncomeTransactionDto {
    id: string;  
    description?: string;
    value: Money;
    dueDate?: string;  
    transactionGroup?: GetIncomeTransactionGroupDto;
}  

File: FinanceApp.Application/Dtos/IncomeTransactionDtos/GetIncomeTransactionDto.cs
using FinanceApp.Application.Dtos.IncomeTransactionGroupDtos;
using FinanceApp.Domain.Entities;

namespace FinanceApp.Application.Dtos.IncomeTransactionDtos;

public class GetIncomeTransactionDto
{
  public Guid Id { get; set; }
  public string Name { get; set; } = string.Empty;
  public string? Description { get; set; }
  public Money Value { get; set; } = new();
  public DateTimeOffset? DueDate { get; set; }
  public GetIncomeTransactionGroupDto? TransactionGroup { get; set; }
}
